java-squid/src/main/java/org/sonar/java/cfg/CFG.java

```diff
@@ -803,7 +834,11 @@ private Block createUnconditionalJump(Tree terminator, @Nullable Block target) {
     Block result = createBlock();
     result.terminator = terminator;
     if (target != null) {
-      result.addSuccessor(target);
+      if (target == exitBlock()) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1698)